### PR TITLE
Implement FastAPI Unit Tests

### PR DESCRIPTION
**Pull request by NeonDaniel**
# Description
Adds test coverage for all exposed endpoints with mocked backend methods
Addresses some edge-case bugs in `request.client` object handling
Fixes/completes model type annotations

# Issues

# Other Notes

### CODE REVIEW

File: neon_hana/app/routers/util.py [hunk 1]
```diff
@@ -34,11 +34,17 @@
 
 @util_route.get("/client_ip", response_class=PlainTextResponse)
 async def api_client_ip(request: Request) -> str:
-    client_manager.validate_auth("", request.client.host)
-    return request.client.host
+    ip_addr = request.client.host if request.client else "127.0.0.1"
+    if len(ip_addr.split('.')) != 4:
```
Comment (mikejgray) [inline neon_hana/app/routers/util.py]: nit(non-blocking): There is an outside chance a hostname could have multiple subdomains and end up triggering this conditional. Might be worth using regex, but this won't make or break the PR IMHO

File: requirements/test_requirements.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
 pytest
 mock
+httpx
```
Comment (mikejgray) [inline requirements/test_requirements.txt:3]: I didn't see this used - was it an initial implementation but removed later?

**Review by mikejgray — APPROVED** [review]

File: requirements/test_requirements.txt (hunk 2 above)
Comment (NeonDaniel) [reply requirements/test_requirements.txt:3]: It's required for the FastAPI TestClient, according to [FastAPI documentation](https://fastapi.tiangolo.com/tutorial/testing/#using-testclient)

File: neon_hana/app/routers/util.py (hunk 1 above)
Comment (NeonDaniel) [reply neon_hana/app/routers/util.py]: Found a rx string and added a check with some unit tests to better evaluate this. I forgot to go back through this; I didn't realize the host param could be a hostname until it came up in GHA tests
